### PR TITLE
DOC: improve documentation of singular value decomposition

### PR DESCRIPTION
**Pull request by dw61**
`u @ np.diag(s) @ vh` can only reproduce the original matrix when `full_matrices` is `False`, otherwise dimension does not match on `np.diag(s)`, with either the square matrix `u` or the square matrix `vh`, whichever is larger.

I tested it like this.
```python3
A = np.array([[1, 0, 0, 0, 2], [0, 0, 3, 0, 0], [0, 0, 0, 0, 0], [0, 2, 0, 0, 0]])
U, S, VT = np.linalg.svd(A, full_matrices=False)
print(U.shape, S.shape, VT.shape)
U @ np.diag(S) @ VT
```
By setting `full_matrices=False`, this code works, and prints
```python
(4, 4) (4,) (4, 5)
array([[1., 0., 0., 0., 2.],
       [0., 0., 3., 0., 0.],
       [0., 0., 0., 0., 0.],
       [0., 2., 0., 0., 0.]])
```

But by default, `full_matrices=True`, so
```python
A = np.array([[1, 0, 0, 0, 2], [0, 0, 3, 0, 0], [0, 0, 0, 0, 0], [0, 2, 0, 0, 0]]).T
U, S, VT = np.linalg.svd(A)
print(U.shape, S.shape, VT.shape)
U @ np.diag(S) @ VT
```
returns a value error and prints
```python
(4, 4) (4,) (5, 5)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-25-da4215a3b044> in <module>
      2 U, S, VT = np.linalg.svd(A)
      3 print(U.shape, S.shape, VT.shape)
----> 4 U @ np.diag(S) @ VT

ValueError: shapes (4,4) and (5,5) not aligned: 4 (dim 1) != 5 (dim 0)
```

I hope this clarifies some confusion when reproducing the matrix.

### CODE REVIEW

File: numpy/linalg/linalg.py [hunk 1]
```diff
@@ -1478,8 +1478,9 @@ def svd(a, full_matrices=True, compute_uv=True, hermitian=False):
     """
     Singular Value Decomposition.
 
-    When `a` is a 2D array, it is factorized as ``u @ np.diag(s) @ vh
-    = (u * s) @ vh``, where `u` and `vh` are 2D unitary arrays and `s` is a 1D
+    When `a` is a 2D array, and when `full_matrices` is `False`,
```
Comment (charris) [inline numpy/linalg/linalg.py:1481]:
```suggestion
    When `a` is a 2D array and ``full_matrices=False``, then
```

File: numpy/linalg/linalg.py [hunk 2]
```diff
@@ -1478,8 +1478,9 @@ def svd(a, full_matrices=True, compute_uv=True, hermitian=False):
...
+    it is factorized as ``u @ np.diag(s) @ vh = (u * s) @ vh``,
+    where `u` and `vh` are 2D unitary arrays and `s` is a 1D
     array of `a`'s singular values. When `a` is higher-dimensional, SVD is
     applied in stacked mode as explained below.
```
Comment (charris) [inline numpy/linalg/linalg.py:1485]:
```suggestion
    When `a` is a 2D array, and ``full_matrices=False``, then it is
    factorized as ``u @ np.diag(s) @ vh = (u * s) @ vh``, where
    `u` and the Hermitian transpose of `vh` are 2D arrays with
    orthonormal columns and `s` is a 1D array of `a`'s singular
    values. When `a` is higher-dimensional, SVD is applied in
    stacked mode as explained below.
```

**Comment by charris** [comment]
Thanks @dw61 .

**Comment by InessaPawson** [comment]
Hi-five on merging your first pull request to NumPy, @dw61! We hope you stick around! Your choices aren’t limited to programming – you can review pull requests, help us stay on top of new and old  issues, develop educational material, work on our website, add or improve graphic design, create marketing materials, translate website content, write grant proposals, and help with other fundraising initiatives. For more info, check out: https://numpy.org/contribute
Also, consider joining our mailing list. This is a great way to connect with other cool people in our community and be part of important conversations that affect the development of NumPy: https://mail.python.org/mailman/listinfo/numpy-discussion

**Pull request merged**